src/main/webapp/app/exercises/programming/shared/code-editor/ace/code-editor-ace.component.ts

```diff
@@ -108,26 +117,25 @@ export class CodeEditorAceComponent implements AfterViewInit, OnChanges, OnDestr
     annotationsArray: Array<Annotation> = [];
     annotationChange: Subscription;
     fileSession: FileSession = {};
-    // Inline feedback variables
-    fileFeedbacks: Feedback[];
-    fileFeedbackPerLine: { [line: number]: Feedback } = {};
-    linesWithNewFeedback: number[] = []; // lines with new feedback, which is not yet saved, but must already be displayed to be editable
+    newFeedbackLines: number[] = []; // new feedback, which is not yet saved, but must already be displayed to be editable
     editorSession: any;
     markerIds: number[] = [];
     gutterHighlights: Map<number, string[]> = new Map<number, string[]>();
     tabSize = 4;
 
+    // Detecting editor resize events
+    resizeObserver: ResizeObserver;
+
     // Icons
     readonly faPlusSquare = faPlusSquare;
     readonly faCircleNotch = faCircleNotch;
 
     /**
-     * Get all line numbers in the current editor session that have inline feedback or new feedback (which is only shown temporarily)
+     * Filter feedback: Only referenced to current file
      */
-    get linesWithInlineFeedbackShown(): number[] {
-        const lines = this.linesWithNewFeedback.concat(Object.keys(this.fileFeedbackPerLine).map((line) => parseInt(line)));
-        lines.sort((a, b) => a - b);
-        return lines;
+    filterFeedbackForFile(feedbacks: Feedback[]): Feedback[] {
+        if (!this.selectedFile) return [];
```

Comment:
```suggestion
        if (!this.selectedFile) {
            return [];
        }
```
style nitpick
(I would have assumed prettier would also enforce braces for blocks similar to how Spotless does it for the server code.)